pkg/oc/cli/admin/release/extract.go

```diff
@@ -0,0 +1,86 @@
+package release
+
+import (
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	digest "github.com/opencontainers/go-digest"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/genericclioptions"
+
+	"github.com/openshift/origin/pkg/image/apis/image/docker10"
+	imagereference "github.com/openshift/origin/pkg/image/apis/image/reference"
+	"github.com/openshift/origin/pkg/oc/cli/image/extract"
+)
+
+func NewExtract(f kcmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := &extractOptions{
+		Out:       streams.Out,
+		ErrOut:    streams.ErrOut,
+		Directory: ".",
+	}
+	cmd := &cobra.Command{
+		Use:   "extract",
+		Short: "Extract the contents of an update payload to disk",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := o.Complete(cmd, args); err != nil {
+				return err
+			}
+			return o.Run()
+		},
+	}
+	flags := cmd.Flags()
+	flags.StringVar(&o.Directory, "to", o.Directory, "Directory to write release contents to, defaults to the current directory.")
+	flags.StringVar(&o.From, "from", o.From, "Image containing the release payload.")
+	return cmd
+}
+
+type extractOptions struct {
+	Out, ErrOut io.Writer
+
+	Directory string
+	From      string
+}
+
+func (o *extractOptions) Complete(cmd *cobra.Command, args []string) error {
+	return nil
+}
+
+func (o *extractOptions) Run() error {
+	if len(o.From) == 0 {
+		return fmt.Errorf("must specify an image to extract the release payload from")
+	}
+
+	dir := o.Directory
+
+	src := o.From
+	ref, err := imagereference.Parse(src)
+	if err != nil {
+		return err
+	}
+	opts := extract.NewOptions(genericclioptions.IOStreams{Out: o.Out, ErrOut: o.ErrOut})
```

Comment:
This will be simpler if you embed `genericclioptions.IOStreams`, like mentioned before.